### PR TITLE
Snowfakery uses load files

### PR DESCRIPTION
**Pull request by prescod**
Changes

Snowfakery can now use "load files" to drive the load process.

Mapping file params like "Serial", "rest, "bulk" are now case-insensitive.

### CODE REVIEW

File: docs/data.rst [hunk 1]
```diff
@@ -641,6 +641,51 @@ then when you run the command above it will run the recipe
...
+  bulk_mode: parallel
+```
+
+Which would specifically load accounts using the bulk API's parallel mode.
+
+The specific keys that you can associate with an object are:
+
+* api: "smart", "rest" or "bulk"
+* batch_size: a number
+* bulk_mode: "serial" or "parallel"]
```
Comment (davisagli) [inline docs/data.rst:669]:
```suggestion
* bulk_mode: "serial" or "parallel"
```

File: docs/data.rst [hunk 2]
```diff
@@ -641,6 +641,51 @@ then when you run the command above it will run the recipe
...
+* load_after: the name of another sobject to wait for before loading
```
Comment (davisagli) [inline docs/data.rst:670]: These docs would be more useful if you explained what the options do / when to use them -- but I haven't checked to see what is covered already in the context of mapping files.

File: cumulusci/tasks/bulkdata/mapping_parser.py [hunk 3]
```diff
@@ -86,6 +100,12 @@ class MappingStep(CCIDictModel):
     ] = None  # default should come from task options
     anchor_date: Optional[Union[str, date]] = None
 
+    @validator("bulk_mode", "api", "action", pre=True)
+    def case_normalize(cls, val):
+        if isinstance(val, Enum):
+            return val
+        return ENUM_VALUES.get(val.lower())
```
Comment (davisagli) [inline cumulusci/tasks/bulkdata/mapping_parser.py:107]: I tested with an invalid value, and the error is pretty inscrutable:

```
2021-03-31 17:12:10: Getting org info from Salesforce CLI for test-fnuu4th1ktiu@example.com
2021-03-31 17:12:14: Beginning task: GenerateAndLoadDataFromYaml
2021-03-31 17:12:14: As user: test-fnuu4th1ktiu@example.com
2021-03-31 17:12:14: In org: 00D21000000JHHa
2021-03-31 17:12:14: 
2021-03-31 17:12:14: Beginning task: GenerateDataFromYaml
2021-03-31 17:12:14: 

1 validation error for SObjectRuleDeclarationFile
__root__ -> 0 -> api
  unexpected value; permitted: 'smart', 'rest', 'bulk' (type=value_error.const; given=bulko; permitted=('smart', 'rest', 'bulk'))
```

File: requirements/prod.txt [hunk 4]
```diff
@@ -116,7 +116,7 @@ six==1.15.0
     #   fs
     #   python-dateutil
     #   salesforce-bulk
-snowfakery==1.7
+snowfakery==1.8
```
Comment (davisagli) [inline requirements/prod.txt:119]: update

**Review by davisagli — CHANGES_REQUESTED** [review]

File: cumulusci/core/utils.py [hunk 5]
```diff
@@ -98,7 +98,7 @@ def process_glob_list_arg(arg):
     return list(dict.fromkeys(files))
 
 
-def process_list_arg(arg):
+def process_list_arg(arg) -> T.Optional[T.List[str]]:
```
[the remark this answers is not in the document]
Comment (prescod) [reply cumulusci/core/utils.py]: Removed.

File: cumulusci/tasks/bulkdata/generate_from_yaml.py [hunk 6]
```diff
@@ -52,6 +53,9 @@ class GenerateDataFromYaml(BaseGenerateDataTask):
         "working_directory": {
             "description": "Default path for temporary / working files"
         },
+        "loading_rules": {
+            "description": "Path to .load.yml containing rules to use to load the file. Defaults to <recipename>.load.yml . Multiple flles can be comma separated."
```
[the remark this answers is not in the document]
Comment (prescod) [reply cumulusci/tasks/bulkdata/generate_from_yaml.py]: I use the term "recipe" more consistently now.

File: docs/data.rst (hunk 1 above)
Comment (prescod) [reply docs/data.rst]: Fixed.

File: docs/data.rst (hunk 2 above)
Comment (prescod) [reply docs/data.rst:670]: Link added.

File: cumulusci/tasks/bulkdata/mapping_parser.py (hunk 3 above)
Comment (prescod) [reply cumulusci/tasks/bulkdata/mapping_parser.py:107]: I'll make a WI to improve this in the future as we discussed.

File: requirements/prod.txt (hunk 4 above)
Comment (prescod) [reply requirements/prod.txt]: Updated.

**Review by Br4nd0R — APPROVED** [review]

**Review by davisagli — APPROVED** [review]

**Pull request merged**